### DISCUSSION TITLE
EventToCommandBehavior EventName and type safety

### DISCUSSION THREAD

**Discussion by LukasKuchta** [CommunityToolkit/Maui#726] [Q&A]
Hello, is there any way to make EventName parametr type safety? It is string and code compile without error when typo is present. But function is not working. 

```
   <toolkit:EventToCommandBehavior
                EventName="Clicked"
                Command="{Binding MyCustomCommand}" />
```

Any advice ? Thanks L.

**Comment by mhrastegari** [comment]
One way is to create your own markup extention like i did for font size:

### Extention

```
    public enum AppFontSize 
    {
        Headline1 = 34,
        Headline2 = 24,
        Headline3 = 20,
        Subtitle1 = 16,
        Subtitle2 = 14,
        Body1 = 16,
        Body2 = 14,
        Button = 14,
        Caption = 12,
        Overline = 10,
    }

    public class AppFontSizeExtention : IMarkupExtension<double>
    {
        public AppFontSize Size { get; set; }

        public double ProvideValue(IServiceProvider serviceProvider)
        {
            return (double)Size;
        }

        object IMarkupExtension.ProvideValue(IServiceProvider serviceProvider)
        {
            return ProvideValue(serviceProvider);
        }
    }

```
### Usage

`
            <Label
                FontSize="{resource:AppFontSizeExtention Size=Headline1}"
                HorizontalOptions="StartAndExpand"
                VerticalOptions="CenterAndExpand" />
`

### Learn more
https://learn.microsoft.com/en-us/xamarin/xamarin-forms/xaml/markup-extensions/creating

**Comment by CliffAgius** [comment]
Closed and answered
